crypto/dso/dso_win32.c

```diff
@@ -444,24 +444,20 @@ static char *win32_name_converter(DSO *dso, const char *filename)
     char *translated;
     int len, transform;
 
-    len = strlen(filename);
     transform = ((strstr(filename, "/") == NULL) &&
                  (strstr(filename, "\\") == NULL) &&
                  (strstr(filename, ":") == NULL));
+    /* If transform != 0, then we convert to %s.dll, else just dupe filename */
+
+    len = strlen(filename) + 1;
     if (transform)
-        /* We will convert this to "%s.dll" */
-        translated = OPENSSL_malloc(len + 5);
-    else
-        /* We will simply duplicate filename */
-        translated = OPENSSL_malloc(len + 1);
+        len += strlen(".dll");
```

Comment:
> nit: this could be `sizeof(".dll") - 1`

Yes, `sizeof(string literal) - 1` is a well-known hack for getting string length at compile time. But it's a bit of a misuse of `sizeof()`, and `stlrlen()` is both clearer and not subject to nasty bugs if the string literal is replaced by a string variable

> sizeof(".dll") evaluates to [...] four

Actually, it gives 5 - the `.`, `'d', `'l', `l` and `\0`

Reply:
Thank you both for pointing out this interesting discrepancy, which was new to me.
Now found it also at https://stackoverflow.com/questions/68254687/string-literals-in-sizeof